src/integration_test/java/com/klodnicki/Bike/v2/rest/controller/RentBikeControllerIntegrationTest.java

```diff
@@ -0,0 +1,226 @@
+        bikeRepository.save(bike2NotRented);
+        bikeRepository.save(bike3Rented);
+
+        List<BikeForNormalUserResponseDTO> dtoList = new ArrayList<>();
+        List<Bike> bikeList = new ArrayList<>();
+        bikeList.add(bike1NotRented);
+        bikeList.add(bike2NotRented);
+
+        for (Bike bike : bikeList) {
+            BikeForNormalUserResponseDTO bikeDTO = BikeForNormalUserResponseDTO.builder()
+                    .id(bike.getId())
+                    .serialNumber(bike.getSerialNumber())
+                    .isRented(bike.isRented())
+                    .bikeType(bike.getBikeType())
+                    .build();
+            dtoList.add(bikeDTO);
+        }
+
+        ListBikesForNormalUserResponseDTO expected = new ListBikesForNormalUserResponseDTO(dtoList);
+
+        webTestClient.get()
+                .uri("/api/bikes")
+                .exchange()
+                .expectStatus().isOk()
+                .expectBody(ListBikesForNormalUserResponseDTO.class)
+                .consumeWith(response -> {
+                    ListBikesForNormalUserResponseDTO responseListDTO = response.getResponseBody();
+                    assertNotNull(responseListDTO);
+                    assertFalse(responseListDTO.getListOfBikesDTOs().isEmpty());
+                    assertIterableEquals(expected.getListOfBikesDTOs(), responseListDTO.getListOfBikesDTOs());
+                });
+    }
+
+    @Test
+    public void findBikeForNormalUserById_ShouldMapAndReturnBikeForNormalUserResponseDTO_WhenBikeExistsInDatabaseAndBikeIdIsGiven() {
+        BikeForNormalUserResponseDTO expected = modelMapper.map(bike, BikeForNormalUserResponseDTO.class);
+
+        webTestClient.get()
+                .uri("/api/bikes/" + bike.getId())
+                .exchange()
+                .expectStatus().isOk()
+                .expectBody(BikeForNormalUserResponseDTO.class)
+                .consumeWith(response -> {
+                    BikeForNormalUserResponseDTO bikeResponseDTO = response.getResponseBody();
+                    assertNotNull(bikeResponseDTO);
+                    assertEquals(expected, bikeResponseDTO);
+                });
+    }
+
+    @Test
+    public void rentBike_ShouldInstantiateAndReturnRentResponseDTOAndSaveRentInDatabase_WhenRentRequestIsGiven() {
+
+        bike.setChargingStation(chargingStation);
+        bikeRepository.save(bike);
+        RentRequest rentRequest = new RentRequest(null, user.getId(), bike.getId(), 10);
+
+        webTestClient.post()
+                .uri("/api/rentals/add")
+                .bodyValue(rentRequest)
+                .exchange()
+                .expectStatus().isOk()
+                .expectBody(RentResponseDTO.class)
+                .consumeWith(response -> {
+                    RentResponseDTO responseDTO = response.getResponseBody();
+                    assertNotNull(responseDTO);
+                    assertTrue(rentRepository.count() > 0);
```

Comment:
Should this actually be >0 or ==1? I think the latter is better.

Reply:
but why? both solutions do the same? or not?;)

Reply:
It's due to the fact that if you save more than one time - let's say, two, or ten :) - it's going to be OK according to this test. It just caught my eye, so I commented on that. Let me know if that is worth to correct in your opinion or any ideas regarding that!